checker/checker.go

```diff
@@ -0,0 +1,178 @@
+package checker
+
+//go:generate moq -stub -out checker_mocks_test.go -pkg checker . DeploymentStateGetter SlackNotifier
```

Comment:
We normally put the mocks in their own subdir. This helps to stop the mock polluting the `checker` package with wrongly exported functions and variables.
```suggestion
//go:generate moq -stub -out mock/checker.go -pkg mock . DeploymentStateGetter SlackNotifier
```

Reply:
done